app/src/main/java/com/epfl/beatlink/ui/components/MainComponents.kt

```diff
@@ -420,14 +421,14 @@ fun MusicPlayerUI(api: SpotifyApiViewModel) {
   var currentTrack by remember { mutableStateOf(SpotifyTrack("", "", "", 0, 0, State.PAUSE)) }
   var currentArtist by remember { mutableStateOf(SpotifyArtist("", "", listOf(), 0)) }
 
-  LaunchedEffect(Unit) {
-    api.getPlaybackState { result ->
-      showPlayer = result.isSuccess
-      if (showPlayer) {
-        api.buildAlbum { currentAlbum = it }
-        api.buildTrack { currentTrack = it }
-        api.buildArtist { currentArtist = it }
-      }
+  api.getPlaybackState { result ->
+    showPlayer = result.isSuccess
+    if (showPlayer) {
+      api.buildAlbum { currentAlbum = it }
+      api.buildTrack { currentTrack = it }
+      api.buildArtist { currentArtist = it }
+      Log.d("PLAYER", "PLAYER")
```

Comment:
Check this, it's the same issue as in MapUsersViewModel. You might wanna make it explicit